src/modules/rlm_lua/lua.c

```diff
@@ -718,7 +718,7 @@ static int rlm_lua_get_field(lua_State *L, REQUEST *request, char const *field)
 	return 0;
 }
 
-#define rlm_lua_release_interp(_x)  if (!_x->threads) pthread_mutex_unlock(_x->mutex)
+#define rlm_lua_release_interp(_x)  if (!_x->threads && !_x->mutex) pthread_mutex_unlock(_x->mutex)
```

Comment:
I'm not sure I understand that.  If there's *no* mutex, then it unlocks the mutex?  How does that work?

Reply:
And on top of that, nothing initializes `inst->threads` or `inst->mutex`, so why are those fields even ther?